src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Collect.cs

```diff
@@ -0,0 +1,412 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Entities;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Functions.FunctionArgValidators;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    // Collect(collection:*[...], item1:![...]|*[...], ...)
+    internal class CollectFunction : BuiltinFunction
+    {
+        public override bool AffectsCollectionSchemas => true;
+
+        public bool CanSuggestThisItem => true;
+
+        public override bool SupportsParamCoercion => false;
+
+        public override bool ManipulatesCollections => true;
+
+        public override bool ModifiesValues => true;
+
+        public override bool IsSelfContained => false;
+
+        public override bool RequiresDataSourceScope => true;
+
+        protected virtual bool IsScalar => false;
+
+        public override bool CanSuggestInputColumns => true;
+
+        public override bool MutatesArg0 => true;
+
+        /// <summary>
+        /// Since Arg1 and Arg2 depends on type of Arg1 return false for them.
+        /// </summary>
+        public override bool TryGetTypeForArgSuggestionAt(int argIndex, out DType type)
+        {
+            if (argIndex == 1 || argIndex == 2)
+            {
+                type = default;
+                return false;
```

Comment:
Is this correct? Should it be 0 or 1?

Reply:
This came directly from PA and is unchanged.